### PR TITLE
Picobench cimetrics

### LINKED ISSUE #342: Upload picobench results to CIMetrics
We have CIMetrics comparing the changes to our end-to-end performance tests for each PR, but we're generally ignoring the results of our Picobench benchmarks (though running them for every PR).

We should investigate a way to upload selected entries from each Picobench app to CIMetrics (probably via `--out-fmt=csv`).

### PR DESCRIPTION
**Pull request by eddyashton**
The picobenchmarks in `ctest` now output their results to a csv file. This file is parsed in the `push_perf_data` task in our CI, where certain values are extracted and pushed to CIMetrics to be graphed in each PR.

This resolves #342.

### CODE REVIEW

**Comment by jumaffre** [comment]
As discussed, this will not yet work because of a known limitation of cimetrics (https://github.com/jumaffre/cimetrics/issues/31). I will fix this in cimetrics and release a new version to get this working.

File: .azure-pipelines-templates/push_perf_data.yml [hunk 1]
```diff
@@ -19,6 +19,13 @@ steps:
     workingDirectory: build
 
 # cimetrics related tasks: render and publish results for Pull Requests build
+- script: ./cimetrics_env.sh python upload_pico_metrics.py
```
Comment (achamayou) [inline .azure-pipelines-templates/push_perf_data.yml]: I think we always want to upload, it's just the plotting/posting we only want to do on PR.

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Comment by cimetrics** [comment]
images

**Pull request merged**